repo/repo.go

```diff
@@ -217,17 +248,27 @@ func parseRepositoryDirectives(directives []rule.Directive) (repos []*rule.Rule,
 	return repos, nil
 }
 
+type RepoMacroDirective struct {
```

Comment:
I really like making this a struct:
nit: Calling this a "Directive" is a bit confusing, since this is very specific to a `gazelle:repository_macro statement`. I think calling the struct a `RepoMacro` would be more accurate.